harness/determined/deploy/local/cluster_utils.py

```diff
@@ -1,28 +1,51 @@
 import os
+import pathlib
 import re
 import socket
 import subprocess
 import sys
 import tempfile
-from pathlib import Path
-from typing import Any, Dict, List, Optional, Sequence
+import threading
+from typing import Any, Dict, List, Optional, Sequence, Type
 
 import appdirs
 import docker
 
-import determined
-import determined.deploy
 from determined.common import yaml
 from determined.deploy.errors import MasterTimeoutExpired
 from determined.deploy.healthcheck import wait_for_master
 
 AGENT_NAME_DEFAULT = f"det-agent-{socket.gethostname()}"
 MASTER_PORT_DEFAULT = 8080
+DB_NAME = "determined-db"
+NETWORK_NAME = "determined-network"
+VOLUME_NAME = "determined-db-volume"
 
 # This object, when included in the host config in a container creation request, tells Docker to
 # expose all host GPUs inside a container.
 GPU_DEVICE_REQUEST = {"Driver": "nvidia", "Count": -1, "Capabilities": [["gpu", "utility"]]}
 
+Container = Type[docker.models.containers.Container]
+
+# These defaults come from master/packaging/master.yaml (except for host_path).
+MASTER_CONF_DEFAULT = {
+    "db": {
+        "user": "postgres",
+        "host": "determined-db",
+        "port": 5432,
+        "name": "determined",
+    },
+    "checkpoint_storage": {
+        "type": "shared_fs",
+        "host_path": appdirs.user_data_dir("determined"),
+        "save_experiment_best": 0,
+        "save_trial_best": 1,
+        "save_trial_latest": 1,
+    },
+}
+
+docker_client = docker.from_env()
```

Comment:
is this going to cause side-effects for people using e.g. `det deploy aws` and who don't have docker installed?

Honestly, even if it doesn't (if `det.deploy.local` isn't imported by `det deploy aws`), I would rather have explicit protection.  A singleton is nice but offer some lazy initialization:

```python
_docker_client = None

def docker_client():
  global _docker_client
  if not _docker_client:
    _docker_client = docker.from_env()
  return _docker_client
```